wiki/osu!wiki_contribution_guide/en.md

```diff
@@ -1,178 +1,909 @@
+upstream        https://github.com/ppy/osu-wiki.git (push)
+```
+
+##### Default terminal text-editor
+
+Reference: <http://web.mit.edu/6.005/www/sp14/tutorial/git/config.html>
+
+To do a quick test, type in ``nano``/``vim``/``emacs`` in Git Bash or terminal, and see what happens.
+If the terminal changed rather than an error given, then Git is most likely using this terminal text-editor by default.
+Read what was given for the quit command.
+
+- By default, you will be using ``nano`` if you are in Linux environment.
+- If you are in Windows environment or your Linux environment does not provide ``nano`` by default, then ``vim`` will be used instead.
+- For Mac OSX environment, ``emacs`` should be set by default if it is not ``nano``, or ``vim`` will be used instead.
+
+If you have an adversity to terminal text editor, and want to use a familiar text editor like Notepad++ or GitHub's Atom instead, [swcarpentry provides a guide to set Git to always use that instead][Git set default editor].
+
+If you are fine with the given terminal text editor, then you can skip the above configuration and read below.
+If you already set your own, ignore below and head to the next header.
+
+---
+
+For ``nano``, it is very self-explanatory as all the important commands was shown at the lower part of the editor.
+
+Press these two keys, ``Ctrl``-``X``, together at the same time to exit ``nano`` editor.
+If the content was changed, the lower-part will be changed to ask for confirmation; press ``Y`` key to save changes, or press ``N`` key to cancel changes.
+
+---
+
+![vim interface](./img/Contribute_vim.jpg "vim interface")
+
+For ``vim``, it has its own set of keyboard commands, and most of it can be ignored unless you are using it to write files.
+
+The only three keywords you need to remember when using ``vim`` appears are as follow:
+
+```
+1. i   (lowercase i key, once)
+2. Esc
+3. ZZ  (UPPERCASE Z key, twice) or :q (a colon and a lowercase q key)
+```
+
+By default, you will be in normal mode.
+
+- The first command tells ``vim`` to go to insert mode, so you can write on it.
+  - If you do not want to modify the text content, you can directly skip to the third command if the first command was not used.
+- The second command tells ``vim`` to go back to normal mode, which allows you to exit.
+- The third command tells ``vim`` to save and exit if some text was given; exit and abort the commit if no text was given.
+
+Oh, and if you got the [E325 error][E325 error], do this:
+
+```prolog
+rm .git/.COMMIT_EDITMSG.swp
+```
+
+#### Accessing your local copy
+
+Open up Git Bash or your terminal of choice (if you have not), and type in this command in the CLI:
+
+```prolog
+git status
+```
+
+This command is used to check for Git's response about the status of the local copy in this directory.
+
+If it throws up an error (meaning, you are in the wrong directory), type in this command in the CLI:
+
+```prolog
+cd $HOME
```

Comment:
Pretty much a few sentences below, the `cd ~` shorthand is used. I think that for consistency on the guide, one or the other should be used.